packages/astro/src/core/build/plugins/plugin-middleware.ts

```diff
@@ -1,20 +1,17 @@
+import type { Plugin as VitePlugin } from 'vite';
 import { vitePluginMiddlewareBuild } from '../../middleware/vite-plugin.js';
 import type { BuildInternals } from '../internal.js';
-import type { AstroBuildPlugin } from '../plugin.js';
 import type { StaticBuildOptions } from '../types.js';
 
 export function pluginMiddleware(
 	opts: StaticBuildOptions,
 	internals: BuildInternals,
-): AstroBuildPlugin {
+): VitePlugin {
+	const plugin = vitePluginMiddlewareBuild(opts, internals);
 	return {
-		targets: ['server'],
-		hooks: {
-			'build:before': () => {
-				return {
-					vitePlugin: vitePluginMiddlewareBuild(opts, internals),
-				};
-			},
+		...plugin,
+		applyToEnvironment(environment) {
+			return environment.name === 'ssr';
```

Comment:
We do want run the middleware in pretender pages too. Should we add `prerender` environment too?

Reply:
Yes we should, will update this.